Python/Product/PythonTools/PythonTools/Intellisense/ProjectAnalyzer.cs

```diff
@@ -1960,6 +1974,66 @@ class ExprWalker : PythonWalker {
             return new AnalysisVariable(type, location);
         }
 
+        internal static string ExpressionForDataTip(IServiceProvider serviceProvider, ITextView view, SnapshotSpan span) {
```

Comment:
If you _do_ port that over, it needs rewriting to avoid using `ThreadHelper`. We need to pass around `IServiceProvider` instances and use those so that we can mock everything out correctly for tests.